File: frontend/src/app/utils/lesson-status.ts
/**
 * Status przebiegu dowolnych zajęć w ramach kursu nauki jazdy.
 */
export enum LessonStatus {

  REQUESTED,
  ACCEPTED,
  REJECTED,
  PASSED,
  FAILED

}

export namespace LessonStatus {

  const REQUESTED_TRANSLATION = "Zgłoszenie prośby o akceptację";
  const ACCEPTED_TRANSLATION = "Zgłoszenie przyjęte";
  const REJECTED_TRANSLATION = "Zgłoszenie odrzucone";
  const PASSED_TRANSLATION = "Zajęcia zrealizowane";
  const FAILED_TRANSLATION = "Zajęcia niezrealizowane";

  export function translate(status: LessonStatus): string {
    let translator: string;
    switch (LessonStatus[status] as unknown) {
      case LessonStatus.REQUESTED:
        translator = REQUESTED_TRANSLATION;
        break;
      case LessonStatus.ACCEPTED:
        translator = ACCEPTED_TRANSLATION;
        break;
      case LessonStatus.REJECTED:
        translator = REJECTED_TRANSLATION;
        break;
      case LessonStatus.PASSED:
        translator = PASSED_TRANSLATION;
        break;
      case LessonStatus.FAILED:
        translator = FAILED_TRANSLATION;
        break;
      default:
        translator = "";
        break;
    }
    return translator;
  }

}

File: src/main/java/com/project/webapp/drivingschool/data/utils/LessonStatus.java
package com.project.webapp.drivingschool.data.utils;

/**
 * Status przebiegu dowolnych zajęć w ramach kursu nauki jazdy
 */
public enum LessonStatus {

    /**
     * Etap pierwszy:
     * zgłoszenie prośby o akceptację przez prowadzącego
     */
    REQUESTED,

    /**
     * Etap drugi:
     * zgłoszenie zostaje przyjęte i czeka na realizację
     */
    ACCEPTED,

    /**
     * Etap drugi:
     * zgłoszenie zostaje odrzucone
     */
    REJECTED,

    /**
     * Etap trzeci:
     * zgłoszenie zostaje zrealizowane (zajęcia odbywają się)
     */
    PASSED,

    /**
     * Etap trzeci:
     * zgłoszenie nie zostaje zrealizowane (zajęcia nie odbywają się)
     */
    FAILED;

    /**
     * Sprawdzenie, czy wydarzenie o podanym statusie
     * jest wydarzeniem aktualnym tzn. nieodrzuconym.
     *
     * @return true - jeśli wydarzenie aktualne, false - w przeciwnym razie
     */
    public boolean isActual() {
        return LessonStatus.ACCEPTED.equals(this) || LessonStatus.PASSED.equals(this)
                || LessonStatus.FAILED.equals(this);
    }

}
